scripts/main/album.js

```diff
@@ -825,79 +894,49 @@ album.setPublic = function (albumID, e) {
+			$('.basicModal .switch input[name="is_nsfw"]').prop("checked", false);
 		}
 
-		if (album.json.public === "1") {
-			$('.basicModal .switch input[name="public"]').click();
+		if (album.json.is_public) {
+			$('.basicModal .switch input[name="is_public"]').click();
 		} else {
 			$(".basicModal .choice input").attr("disabled", true);
 		}
 
-		$('.basicModal .choice input[name="password"]').on("change", function () {
+		$('.basicModal .choice input[name="has_password"]').on("change", function () {
 			if ($(this).prop("checked") === true) $('.basicModal .choice input[name="passwordtext"]').show().focus();
 			else $('.basicModal .choice input[name="passwordtext"]').hide();
 		});
-
-		return true;
 	}
 
 	albums.refresh();
 
 	// Set public
-	if ($('.basicModal .switch input[name="nsfw"]:checked').length === 1) {
-		album.json.nsfw = "1";
-	} else {
-		album.json.nsfw = "0";
-	}
+	album.json.is_nsfw = $('.basicModal .switch input[name="is_nsfw"]:checked').length === 1;
 
 	// Set public
-	if ($('.basicModal .switch input[name="public"]:checked').length === 1) {
-		album.json.public = "1";
-	} else {
-		album.json.public = "0";
-	}
+	album.json.is_public = $('.basicModal .switch input[name="is_public"]:checked').length === 1;
 
 	// Set full photo
-	if ($('.basicModal .choice input[name="full_photo"]:checked').length === 1) {
-		album.json.full_photo = "1";
-	} else {
-		album.json.full_photo = "0";
-	}
+	album.json.grants_full_photo = $('.basicModal .choice input[name="grants_full_photo"]:checked').length === 1;
 
 	// Set visible
-	if ($('.basicModal .choice input[name="hidden"]:checked').length === 1) {
-		album.json.visible = "0";
-	} else {
-		album.json.visible = "1";
-	}
+	album.json.requires_link = $('.basicModal .choice input[name="requires_link"]:checked').length === 1;
 
 	// Set downloadable
-	if ($('.basicModal .choice input[name="downloadable"]:checked').length === 1) {
-		album.json.downloadable = "1";
-	} else {
-		album.json.downloadable = "0";
-	}
+	album.json.is_downloadable = $('.basicModal .choice input[name="is_downloadable"]:checked').length === 1;
 
 	// Set share_button_visible
-	if ($('.basicModal .choice input[name="share_button_visible"]:checked').length === 1) {
-		album.json.share_button_visible = "1";
-	} else {
-		album.json.share_button_visible = "0";
-	}
+	album.json.is_share_button_visible = $('.basicModal .choice input[name="is_share_button_visible"]:checked').length === 1;
 
 	// Set password
-	let oldPassword = album.json.password;
-	if ($('.basicModal .choice input[name="password"]:checked').length === 1) {
-		password = $('.basicModal .choice input[name="passwordtext"]').val();
-		album.json.password = "1";
-	} else {
-		password = "";
-		album.json.password = "0";
-	}
+	// TODO: IMHO the following line does not make much sense. The JSON has no attribute `password`, because the server does not send it. Thus `oldPassword` is always undefined.
```

Comment:
I remember that there was some distinction between:
- removing the password
- updating the password
- keeping the password unchanged.

Reply:
I just checked with the 4.4.0 release and `password` attribute _is_ present there. So this looks like a code that wasn't updated when #271 got merged, but it probably should've been.